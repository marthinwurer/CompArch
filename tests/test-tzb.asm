; test the tzb instruction
	.entry	start
;
	.org	0x20
zero:	.word	0
one:	.word	0x00001
five:	.word	0x00005
neg1:	.word	0xfffff
neg5:	.word	0xffffb
error:	.word	0x00bad
;
	.org	0x100
start:
	ldb	zero
	tzb	test2
err1:	ldb	error
;
test2:	ldb	one
	tzb	err3
	tr	test3
err2:	ldb	error
;
test3:	ldb	five
	tzb	err3
	tr	test4
err3:	ldb	error
;
test4:	ldb	neg1
	tzb	err4
	tr	test5
err4:	ldb	error
;
test5:	ldb	zero
	tzb	test6
err5:	ldb	error
;
test6:	ldb	neg5
	tzb	err6
	tr	out
err6:	ldb	error
;
out:
	halt
