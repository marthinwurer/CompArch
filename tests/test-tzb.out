CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 100
100:  11020  LDB   020  A[00000]  B[00000]  XR[000]
101:  33103  TZB   103  A[00000]  B[00000]  XR[000]
103:  11021  LDB   021  A[00000]  B[00000]  XR[000]
104:  3310a  TZB   10a  A[00000]  B[00001]  XR[000]
105:  30107  TR    107  A[00000]  B[00001]  XR[000]
107:  11022  LDB   022  A[00000]  B[00001]  XR[000]
108:  3310a  TZB   10a  A[00000]  B[00005]  XR[000]
109:  3010b  TR    10b  A[00000]  B[00005]  XR[000]
10b:  11023  LDB   023  A[00000]  B[00005]  XR[000]
10c:  3310e  TZB   10e  A[00000]  B[fffff]  XR[000]
10d:  3010f  TR    10f  A[00000]  B[fffff]  XR[000]
10f:  11020  LDB   020  A[00000]  B[fffff]  XR[000]
110:  33112  TZB   112  A[00000]  B[00000]  XR[000]
112:  11024  LDB   024  A[00000]  B[00000]  XR[000]
113:  33115  TZB   115  A[00000]  B[ffffb]  XR[000]
114:  30116  TR    116  A[00000]  B[ffffb]  XR[000]
116:  00000  HALT       A[00000]  B[ffffb]  XR[000]

Machine Halted - HALT instruction executed

Simulated time 90 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
